cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java

```diff
@@ -358,20 +393,53 @@ private static void writeBatteries(Network network, String cimNamespace, XMLStre
     }
 
     private static <I extends ReactiveLimitsHolder & Injection<I>> String obtainOperatingMode(I i, double minP, double maxP, double targetP) {
-        String kind = i.getProperty(Conversion.PROPERTY_CGMES_SYNCHRONOUS_MACHINE_TYPE);
-        String operatingMode = i.getProperty(Conversion.PROPERTY_CGMES_SYNCHRONOUS_MACHINE_OPERATING_MODE);
-        String calculatedKind = CgmesExportUtil.obtainCalculatedSynchronousMachineKind(minP, maxP, CgmesExportUtil.obtainCurve(i), kind);
-        String calculatedOperatingMode = obtainOperatingMode(targetP);
-        return operatingMode != null && calculatedKind.contains(operatingMode) ? operatingMode : calculatedOperatingMode;
+        String calculatedKind = obtainCalculatedSynchronousMachineKind(minP, maxP, obtainCurve(i), i instanceof Generator gen && gen.isCondenser());
+        return obtainOperatingMode(targetP, i, calculatedKind);
     }
 
-    private static String obtainOperatingMode(double targetP) {
+    private static String obtainOperatingMode(double targetP, Injection<?> injection, String calculatedKind) {
         if (targetP < 0) {
-            return "motor";
+            return OPERATING_MODE_MOTOR;
         } else if (targetP > 0) {
-            return "generator";
+            return OPERATING_MODE_GENERATOR;
         } else {
-            return "condenser";
+            if (isOperatingAsACondenser(injection)) {
+                return OPERATING_MODE_CONDENSER;
+            } else {
+                if (calculatedKind.toLowerCase().contains(OPERATING_MODE_GENERATOR)) {
+                    return OPERATING_MODE_GENERATOR;
+                } else if (calculatedKind.toLowerCase().contains(OPERATING_MODE_MOTOR)) {
+                    return OPERATING_MODE_MOTOR;
+                } else {
+                    return OPERATING_MODE_CONDENSER;
+                }
+            }
+        }
+    }
+
+    private static boolean isOperatingAsACondenser(Injection<?> injection) {
+        switch (injection) {
+            case Generator generator -> {
+                if (generator.isVoltageRegulatorOn()) {
+                    double targetV = generator.getTargetV();
+                    return !Double.isNaN(targetV) && targetV != 0;
+                } else {
+                    double targetQ = generator.getTargetQ();
+                    return !Double.isNaN(targetQ) && targetQ != 0;
+                }
```

Comment:
I suggest a simplification: it isn't necessary to check targetV value since it can't be negative or null because of the checks in the adder and setter.

```suggestion
                return generator.isVoltageRegulatorOn() && !Double.isNaN(generator.getTargetV())
                        || !Double.isNaN(generator.getTargetQ()) && generator.getTargetQ() != 0;
```